spec/requests/media/odk_response_spec.rb

```diff
@@ -35,14 +47,113 @@
       expect(FileUtils.rm(tmp_files)).to be_empty
     end
 
+    it "should safely ignore simple duplicates", database_cleaner: :truncate do
+      # Original
+      submission_file = prepare_and_upload_submission_file("single_question.xml")
+      post submission_path, params: {xml_submission_file: submission_file}, headers: auth_header
+      expect(response).to have_http_status(:created)
+
+      # Duplicate
+      submission_file = prepare_and_upload_submission_file("single_question.xml")
+      post submission_path, params: {xml_submission_file: submission_file}, headers: auth_header
+      expect(response).to have_http_status(:created)
+
+      expect(Response.count).to eq(1)
+
+      form_response = Response.last
+      expect(form_response.form).to eq(form)
+      expect(form_response.answers.count).to eq(1)
+      expect(form_response.odk_xml.byte_size).to be > 0
+
+      tmp_files = Dir.glob(ResponsesController::TMP_UPLOADS_PATH.join("*.xml"))
+      expect(FileUtils.rm(tmp_files)).to be_empty
+    end
+
+    it "should safely ignore parallel duplicates", database_cleaner: :truncate do
+      original = nil
+      duplicate = nil
+
+      original_file = prepare_and_upload_submission_file("nested_group_form_response.xml")
+      duplicate_file = prepare_and_upload_submission_file("nested_group_form_response_duplicate.xml")
+
+      thread1 = Thread.new do
```

Comment:
ah i'm glad with the more complex version of the form you were able to get the test working this way!